presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -437,16 +435,17 @@ protocol::TaskInfo PrestoTask::updateInfoLocked() {
   // TODO(venkatra): Populate these memory stats as well.
   prestoTaskStats.revocableMemoryReservationInBytes = {};
 
+  const int64_t currentBytes = stats.currentBytes;
+
+  const int64_t averageMemoryForLastPeriod =
+      (currentBytes + lastMemoryReservation) / 2;
+
   // Set the lastTaskStatsUpdateMs to execution start time if it is 0.
   if (lastTaskStatsUpdateMs == 0) {
     lastTaskStatsUpdateMs = taskStats.executionStartTimeMs;
   }
 
-  const int64_t currentBytes = stats.currentBytes;
-
-  int64_t averageMemoryForLastPeriod =
-      (currentBytes + lastMemoryReservation) / 2;
-
+  const double sinceLastPeriodMs = currentTimeMs - lastTaskStatsUpdateMs;
```

Comment:
@xiaoxmeng This code is rather fragile. Lots of things happening in a single very long method. Let's consider refactoring.